platform/darwin/src/MGLStyleLayerManager.mm

```diff
@@ -0,0 +1,84 @@
+#import "MGLStyleLayerManager.h"
+
+#import "MGLBackgroundStyleLayer_Private.h"
+#import "MGLCircleStyleLayer_Private.h"
+#import "MGLFillExtrusionStyleLayer_Private.h"
+#import "MGLFillStyleLayer_Private.h"
+#import "MGLHeatmapStyleLayer_Private.h"
+#import "MGLHillshadeStyleLayer_Private.h"
+#import "MGLLineStyleLayer_Private.h"
+#import "MGLRasterStyleLayer_Private.h"
+#import "MGLSymbolStyleLayer_Private.h"
+#import "MGLOpenGLStyleLayer_Private.h"
+
+#include <vector>
+
+namespace mbgl {
+
+LayerManagerDarwin::LayerManagerDarwin() {
+    addLayerType(std::make_unique<FillStyleLayerPeerFactory>());
+    addLayerType(std::make_unique<LineStyleLayerPeerFactory>());
+    addLayerType(std::make_unique<CircleStyleLayerPeerFactory>());
+    addLayerType(std::make_unique<SymbolStyleLayerPeerFactory>());
+    addLayerType(std::make_unique<RasterStyleLayerPeerFactory>());
+    addLayerType(std::make_unique<BackgroundStyleLayerPeerFactory>());
+    addLayerType(std::make_unique<HillshadeStyleLayerPeerFactory>());
+    addLayerType(std::make_unique<FillExtrusionStyleLayerPeerFactory>());
+    addLayerType(std::make_unique<HeatmapStyleLayerPeerFactory>());
+    addLayerType(std::make_unique<OpenGLStyleLayerPeerFactory>());
+}
+
+LayerManagerDarwin::~LayerManagerDarwin() = default;
+
+MGLStyleLayer* LayerManagerDarwin::createPeer(style::Layer* layer) {
+    auto* typeInfo = layer->getTypeInfo();
+    assert(typeInfo);
+    for (const auto& factory: factories) {
+        if (factory->getCoreLayerFactory()->getTypeInfo() == typeInfo) {
+            return factory->createPeer(layer);
+        }
+    }
+    assert(false);
```

Comment:
There are some differences between C++ `assert()` and Objective-C `NSCAssert()`. If I remember correctly, the former produces inscrutable crash logs, whereas the latter is disabled in Release builds. Darwin-specific code generally uses `NSCAssert()` or the Mapbox-specific `MGLCAssert()` that @fabian-guerra added in #13235.